modules/graph/utils/thread_group.h

```diff
@@ -31,24 +33,66 @@ class ThreadGroup {
   using return_t = Status;
 
  public:
+  explicit ThreadGroup(tid_t parallelism = std::numeric_limits<tid_t>::max())
```

Comment:
The `parallelism` obviously cannot be `MAX_INT`,  would `std::thread::hardware_concurrency` be better?